sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1687,6 +1687,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_PASS_PARTITION_BY_AS_OPTIONS =
+    buildConf("spark.sql.legacy.sources.write.passPartitionByAsOptions")
+      .internal()
+      .doc("Whether to pass the partitionBy columns as options in DataFrameWriter." +
+        " Data source V1 now silently drops partitionBy columns for non-file-format sources;" +
+        " turning the flag on provides a way for these sources to see these partitionBy columns.")
```

Comment:
nit: the format in this file should be `"text " + "more text"` instead of `"text" + " more text"`